internal/compiler/output_columns.go

```diff
@@ -213,12 +213,26 @@ func outputColumns(qc *QueryCatalog, node ast.Node) ([]*Column, error) {
 			}
 			fun, err := qc.catalog.ResolveFuncCall(n)
 			if err == nil {
-				cols = append(cols, &Column{
-					Name:       name,
-					DataType:   dataType(fun.ReturnType),
-					NotNull:    !fun.ReturnTypeNullable,
-					IsFuncCall: true,
-				})
+				if len(fun.Args) > 0 {
+					ref := n.Args.Items[0].(*ast.ColumnRef)
+					columns, err := outputColumnRefs(res, tables, ref)
+					if err != nil {
+						return nil, err
+					}
+					cols = append(cols, &Column{
+						Name:       name,
+						DataType:   columns[0].DataType,
```

Comment:
I think that unfortunately the fix is a lot more complex than this, this is saying that any function with more than 0 args the return type should the type of its first arg?

This is true for most aggregates, (min, max, avg, every) but is not generally true of any function call - in particular `count`, `array_agg` and probably any user-defined aggregate.

https://www.postgresql.org/docs/15/functions-aggregate.html

I think special cases for these common aggregates is probably a useful change.